CMakeLists.txt

```diff
@@ -695,8 +695,27 @@ if(WITH_SYSTEM_NPM)
   if(NOT NPM_EXECUTABLE)
     message(FATAL_ERROR "Can't find npm.")
   endif()
+  # build mgr dashboard frontent requires Angular v15 supports node.js versions: 14.20.x, 16.13.x and 18.10.x
+  # And More information Please See https://angular.io/guide/update-to-version-15#angular-v15-supports-nodejs-versions-1420x-1613x-and-1810x
+  execute_process(
+    COMMAND
+      "sh" "-c"
+      "node -v"
+    OUTPUT_VARIABLE NODE_VERSION
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  string(REPLACE "v" "" NODE_VERSION ${NODE_VERSION})
+  if("${NODE_VERSION}" VERSION_LESS "14.20.0")
+    message(FATAL_ERROR "The node version must greater than version v14.20.0")
+  endif()
+  if(("${NODE_VERSION}" VERSION_GREATER_EQUAL "16.0.0") AND ("${NODE_VERSION}" VERSION_LESS "16.13.0"))
+    message(FATAL_ERROR "The node version must greater than version v16.13.0")
+  endif()
+  if(("${NODE_VERSION}" VERSION_GREATER_EQUAL "18.0.0") AND ("${NODE_VERSION}" VERSION_LESS "18.10.0"))
+    message(FATAL_ERROR "The node version must greater than version v18.10.0")
```

Comment:
I feel that this is going to be tough to maintain in the long run. Perhaps we can just ensure that `npm` is available (as long as the `DASHBOARD_FRONTEND_BUILD` is enabled), and expect that `npm`/`ng` will complain whether an unsupported version is installed?

Reply:
I see the failure message about "The node version must greater than version v14.20.0" in shaman when the operating system is centos 8, it caused by build ceph by ceph.spec, when building the ceph by using ceph.spec, it is default to choice the building the Dashboard UI by using the system npm that is installed by `yum` or `dnf` command, This will have higher requirements on npm version,The situation is consistent with what you said, `I feel that this is going to be tough to maintain in the long run.`, The suggestions about `I think it'd be better to use distro-packages as much as possible` made me think, So I plan to do it in another way, we should build Dashboard UI in make-dist script to support multi-language that using the npm from nodeenv from virtualenv. Dashboard UI is no longer compiled when compiling ceph by ceph.spec.